airflow/models/dag.py

```diff
@@ -462,8 +462,10 @@ def __init__(
         # set timezone from start_date
         tz = None
         if start_date and start_date.tzinfo:
-            tzinfo = None if start_date.tzinfo else settings.TIMEZONE
-            tz = pendulum.instance(start_date, tz=tzinfo).timezone
+            if start_date.tzinfo is None:
+                tz = pendulum.instance(start_date, tz=settings.TIMEZONE).timezone
+            else:
+                tz = pendulum.instance(start_date).timezone
```

Comment:
This seems wrong? The outer `if if start_date and start_date.tzinfo` should already ensure `start_date.tzinfo` is not None.

Reply:
ah, right. I made a mistake. 
Should be 
```
if start_date:
    tzinfo = None if start_date.tzinfo else settings.TIMEZONE
    tz = pendulum.instance(start_date, tz=tzinfo).timezone
    if start_date.tzinfo is None:
        tz = pendulum.instance(start_date, tz=settings.TIMEZONE).timezone
    else:
        tz = pendulum.instance(start_date).timezone
```
but I think we don't need to proceed with this change. 
I will look another issues up.